### PR TITLE
feat: create register validation

### PR DESCRIPTION
**Pull request by hoaingodong1**

### CODE REVIEW

**Comment by hoaingodong1** [comment]
[[BE][Sign up]](https://trello.com/c/QKPdcill/46-besign-up)

File: routes/api.php [hunk 1]
```diff
@@ -17,3 +17,4 @@
 Route::middleware('auth:sanctum')->get('/user', function (Request $request) {
     return $request->user();
 });
+Route::post('/register', [RegisterController::class, 'store']);
```
[the remark this answers is not in the document]
Comment (uica-dev) [reply routes/api.php:20]: @hoaingodong1 Should name the route, then we can dynamically check the validation for create/update/... API with different rules

File: app/Http/Requests/RegisterRequest.php [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    {
+        return true;
+    }
+
+    /**
+     * Get the validation rules that apply to the request.
+     *
+     * @return array<string, mixed>
+     */
+    public function rules()
```
[the remark this answers is not in the document]
Comment (uica-dev) [reply app/Http/Requests/RegisterRequest.php:24]: @hoaingodong1 Dynamic validation check based on the route name

File: app/Http/Requests/RegisterRequest.php [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    {
+        return [
+            'username' => 'required|string|max:20|min:8|unique:accounts',
+            'email' => 'required|email|max:50|unique:accounts',
+            'password' => 'required|min:8|max:50',
+            'password_confirmation' => 'required|same:password',
+            'fullname' => 'required|string|max:255',
+            'avatar_url'=>'required|string',
+            'phone_number' => 'required|string|max:255',
+            'address_line1' => 'required|string|max:255',
```
Comment (uica-dev) [inline app/Http/Requests/RegisterRequest.php:34]: When registering a new account, address info should not be required, they can update later then!

**Review by uica-dev — CHANGES_REQUESTED** [review]

**Pull request closed without merging**